Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Closure
+
+/-!
+# Matroid Circuits
+
+A `Circuit` of a matroid `M` is a minimal set `C` that is dependent in `M`.
+A matroid is determined by its set of circuits, and often the circuits
```

Comment:
Is this already formalized?

Reply:
Yes, but currently just in my repo. It should be part of the next PR, so I thought that including the comment made sense as some general motivation for the notion of a circuit.

Reply:
(I'm assuming by 'this' you are referring to the implied extensionality statement)